packages/platform-core/src/usage-data/account_id_fetcher.ts

```diff
@@ -1,22 +1,32 @@
 import { GetCallerIdentityCommand, STSClient } from '@aws-sdk/client-sts';
 
+const NO_ACCOUNT_ID = '000000000000';
 /**
  * Retrieves the account ID of the user
  */
 export class AccountIdFetcher {
+  private accountId: string | undefined;
   /**
    * constructor for AccountIdFetcher
    */
   constructor(private readonly stsClient = new STSClient()) {}
   fetch = async () => {
-    const stsResponse = await this.stsClient.send(
-      new GetCallerIdentityCommand({})
-    );
-    if (stsResponse && stsResponse.Account) {
-      return stsResponse.Account;
+    if (this.accountId) {
+      return this.accountId;
+    }
+    try {
+      const stsResponse = await this.stsClient.send(
+        new GetCallerIdentityCommand({})
+      );
+      if (stsResponse && stsResponse.Account) {
+        this.accountId = stsResponse.Account;
+        return this.accountId;
+      }
+      // We failed to get the account Id. Most likely the user doesn't have credentials
+      return NO_ACCOUNT_ID;
+    } catch (error) {
+      // We failed to get the account Id. Most likely the user doesn't have credentials
```

Comment:
I saw it's mentioned in the PR description Changes 3, but Why don't we want it to fail?

Reply:
Because this is specifically for usage metrics and not being able to get the accountId for telemetry purposes is not a failure case. In worst case it hides the actual error by throwing this red herring error to customer.